Performance: use virtual scroll container and log level parsing for logs view

MickLesk opened the pull request:
<!--
Note: All PRs with code changes should be targeted to the `dev` branch, pure documentation changes can target `main`
-->

## Proposed change
Replaces the static log display with Angular CDK virtual scroll for improved performance with large log files. Log entries are now parsed to extract log level, enabling level-based styling. Updates LogService to support tailing logs and adjusts tests and component logic accordingly.

<img width="995" height="251" alt="image" src="https://github.com/user-attachments/assets/8a13dd61-ec95-471a-93b7-c813918b597b" />

pnpm build:
<img width="570" height="242" alt="image" src="https://github.com/user-attachments/assets/9564fadf-22a6-4d6b-910b-3eb9a7005de9" />

Tested with local environment:
Loading Time for 3 200MB Logs < 3 sek

<img width="1570" height="839" alt="image" src="https://github.com/user-attachments/assets/9e15b5a1-5445-44cf-b203-6c5d680efc94" />



<!--
⚠️ Important: Pull requests that implement a new feature or enhancement *should almost always target an existing feature request* with evidence of community interest and discussion. This is in order to balance the work of implementing and maintaining new features / enhancements. If that is not currently the case, please open a feature request instead of this PR to gather feedback from both users and the project maintainers.
-->

Closes #11230

## Type of change

- [ ] Bug fix: non-breaking change which fixes an issue.
- [x] New feature / Enhancement: non-breaking change which adds functionality. _Please read the important note above._
- [ ] Breaking change: fix or feature that would cause existing functionality to not work as expected.
- [ ] Documentation only.
- [ ] Other. Please explain:

## Checklist:

<!--
NOTE: PRs that do not address the following will not be merged, please do not skip any relevant items.
-->

- [x] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [x] If applicable, I have included testing coverage for new code in this PR, for [backend](https://docs.paperless-ngx.com/development/#testing) and / or [front-end](https://docs.paperless-ngx.com/development/#testing-and-code-style) changes.
- [x] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [x] If applicable, I have checked that all tests pass, see [documentation](https://docs.paperless-ngx.com/development/#back-end-development).
- [ ] I have run all `pre-commit` hooks, see [documentation](https://docs.paperless-ngx.com/development/#code-formatting-with-pre-commit-hooks).
- [ ] I have made corresponding changes to the documentation as needed.
- [x] I have checked my modifications for any breaking changes.

MickLesk committed:
Add virtual scroll and log level parsing to logs view

Replaces the static log display with Angular CDK virtual scroll for improved performance with large log files. Log entries are now parsed to extract log level, enabling level-based styling. Updates LogService to support tailing logs and adjusts tests and component logic accordingly.

MickLesk committed:
log container use CdkVirtualScrollViewport

Replaces ElementRef with CdkVirtualScrollViewport for the logContainer ViewChild in LogsComponent. Updates scrollToBottom to use the correct method signature and removes an unnecessary scroll mock from the test file.

MickLesk committed:
Add CommonModule and improve log scroll handling

Imported CommonModule in both the logs component and its test to ensure proper Angular functionality. Updated log fetching tests to include the 'tail' parameter. Enhanced scrollToBottom to handle potential errors when the log container is not ready, improving test reliability.

MickLesk committed:
Add 'tail' query parameter to log endpoint

Introduces an optional 'tail' query parameter to limit the number of log lines returned to the last N lines. Updates OpenAPI documentation and adjusts the log reading logic to support this feature.

shamoon commented on src-ui/src/app/components/admin/logs/logs.component.spec.ts:
```diff
@@ -71,7 +74,7 @@ describe('LogsComponent', () => {
       .queryAll(By.directive(NgbNavLink))[1]
       .nativeElement.dispatchEvent(new MouseEvent('click'))
-    expect(logSpy).toHaveBeenCalledWith('mail')
+    expect(logSpy).toHaveBeenCalledWith('mail', { tail: 1000 })
```
At the very least I would go for more than 1000, maybe 5000 if that doesnt affect performance

shamoon reviewed (changes requested):
Additionally the Sonar review suggestions all seem valid and should resolved

https://sonarcloud.io/project/issues?id=paperless-ngx_paperless-ngx&pullRequest=11233&issueStatuses=OPEN,CONFIRMED&sinceLeakPeriod=true

MickLesk committed:
Enhance logs view with auto-refresh interval and infinite scroll

Added an auto-refresh interval selector and improved the logs display with infinite scroll to load older logs on demand. Log entries now show line numbers, and the initial log fetch size was increased. UI and scroll behavior were updated for better usability and performance.

MickLesk committed:
fix liniting check

MickLesk committed:
Reorder imports in logs.component.spec.ts

Adjusted the order of import statements for better organization and consistency in the logs.component.spec.ts test file.

MickLesk committed:
Replace divs with output elements for loading states (SonarCube)

Updated loading indicators in logs.component.html to use <output> elements instead of <div> for improved semantic HTML. Removed unnecessary 'role="status"' attributes from spinner elements.

MickLesk committed:
fix codecov issue with (timing)

MickLesk committed:
revert some funcs

MickLesk committed:
Revert

Updated the logs component to treat logs as simple strings instead of objects with message and level properties. Removed unnecessary parsing and trackBy logic, and adjusted the template and styles accordingly. Also added a scroll method mock in the test setup.

MickLesk commented:
so its minimized, This significantly extends the runtime again. Do with it what you will.

shamoon committed:
Revert "Revert"

This reverts commit d0c69de6a2d98fb390a9165947658454d31e574c.

shamoon committed:
Revert "revert some funcs"

This reverts commit c9daa37e7b989467444dadfcc6caf9d9b1db04d5.

shamoon committed:
Revert "fix codecov issue with (timing)"

This reverts commit 921943bccd636981f9aa041c1696444071fcd1ca.

shamoon committed:
Revert "Replace divs with output elements for loading states (SonarCube)"

This reverts commit 0a4b6c0fbd832097b8fcc140355b90a695ac0a16.

shamoon committed:
Revert "Reorder imports in logs.component.spec.ts"

This reverts commit aa07b606f63fbeddbcaa1e88a1c8bd7f3e7fa5f4.

shamoon committed:
Revert "fix liniting check"

This reverts commit e3e3c147edb2b5dd361ce34c72d32c0509530db1.

shamoon committed:
Revert "Enhance logs view with auto-refresh interval and infinite scroll"

This reverts commit 142930dff7a0134c1005b4530be0a6f40ddd2496.

shamoon committed:
Revert "Add 'tail' query parameter to log endpoint"

This reverts commit 09a1879136a0985fc62c87e6d927a86330210fcd.

shamoon committed:
Revert "Add CommonModule and improve log scroll handling"

This reverts commit 7b3a59603c34fcfa83078572d2090cd33443f410.

shamoon committed:
Fix log container position

shamoon committed:
lint

shamoon committed:
Remove unneeded trackby

shamoon committed:
Merge branch 'dev' into pr/11233

shamoon committed:
Revert last of tail stuff which was across commits

shamoon committed:
Fix imports, remove problematic track

shamoon committed:
Fix missing scrollTo in jest
